### PR TITLE
[DOCS] SQL: Add formal API docs

### LINKED ISSUE #75085: [DOCS] Move SQL APIs to REST APIs section
#75078 adds formal API documentation for the async SQL search APIs. We should add formal API docs for the following APIs:

- [ ] Clear SQL cursor
- [ ] SQL search
- [ ] SQL translate

### PR DESCRIPTION
**Pull request by jrodewig**
Adds formal API docs for the following APIs:

* Clear SQL cursor
* SQL search
* SQL translate

Other changes:

* Removes and redirects the "Supported REST parameters section." This is now covered in the SQL search API docs.
* Updates a few related xrefs.

Closes #75085

### Preview
* SQL APIs: https://elasticsearch_75506.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/sql-apis.html
* Clear SQL cursor: https://elasticsearch_75506.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/clear-sql-cursor-api.html
* SQL search: https://elasticsearch_75506.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/sql-search-api.html
* SQL translate: https://elasticsearch_75506.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/sql-translate-api.html

### CODE REVIEW

File: rest-api-spec/src/main/resources/rest-api-spec/api/sql.clear_cursor.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "sql.clear_cursor":{
     "documentation":{
-      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/current/sql-pagination.html",
+      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/master/clear-sql-cursor-api.html",
```
Comment (sethmlarson) [inline rest-api-spec/src/main/resources/rest-api-spec/api/sql.clear_cursor.json:4]: nit: Let's use `/current/` as is done in other documentation URLs.

**Review by sethmlarson — APPROVED** [review]
URL updates in API specs LGTM with one nit.

**Review by bpintea — COMMENTED** [review]
LG, thank you, I've left some minor notes, just one omission that should be corrected.

File: docs/reference/sql/apis/sql-search-api.asciidoc [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+(Optional, string) Separator for CSV results. Defaults to `,`. The API only
+supports this parameter for CSV responses.
+
+`format`::
+(Optional, string) Format for the response. For valid values, see
+<<sql-rest-format>>.
++
+You can also specify a format using the `Accept` HTTP header. If you specify
+both this parameter and the `Accept` HTTP header, the API only uses this
+parameter.
```
[the remark this answers is not in the document]
Comment (jrodewig) [reply docs/reference/sql/apis/sql-search-api.asciidoc]: Thanks. I'll use:

`If you specify both this parameter and the `Accept` HTTP header, this parameter take precedence.`

File: docs/reference/sql/apis/sql-search-api.asciidoc [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+search. See <<sql-rest-filtering>>.
+
+`index_include_frozen`::
+(Optional, Boolean) If `true`, the search can run on frozen indices. Defaults to
+`false`.
+
+`keep_alive`::
+(Optional, <<time-units,time value>>) Retention period for the search and its
+results. Defaults to `5d` (five days).
+
```
[the remark this answers is not in the document]
Comment (jrodewig) [reply docs/reference/sql/apis/sql-search-api.asciidoc:96]: Thanks for catching this.

File: docs/reference/sql/apis/sql-search-api.asciidoc [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+`filter`::
+(Optional, object) <<query-dsl,Query DSL>> used to filter documents for the SQL
+search. See <<sql-rest-filtering>>.
+
+`index_include_frozen`::
+(Optional, Boolean) If `true`, the search can run on frozen indices. Defaults to
+`false`.
+
+`keep_alive`::
+(Optional, <<time-units,time value>>) Retention period for the search and its
```
[the remark this answers is not in the document]
Comment (jrodewig) [reply docs/reference/sql/apis/sql-search-api.asciidoc]: Thanks for calling this out. For the `keep_alive` parameter, I reworded to:

`Retention period for an async or saved synchronous search. Defaults to `5d` (five days).`

Both the `async` and `saved synchrounous search` copy have xrefs.

I added similar xrefs to the `keep_on_completion` and `wait_for_completion_timeout` parameter definitions.

File: docs/reference/sql/apis/sql-search-api.asciidoc [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+results. Defaults to `5d` (five days).
+
+`page_timeout`::
+(Optional, <<time-units,time value>>) Timeout before a
+<<sql-pagination,pagination request>> fails. Defaults to `45s` (45 seconds).
+
+`params`::
+(Optional, array) Values for variables in the `query`. For variable syntax, see
```
[the remark this answers is not in the document]
Comment (jrodewig) [reply docs/reference/sql/apis/sql-search-api.asciidoc]: Thanks for pointing this out.

File: docs/reference/sql/apis/clear-sql-cursor-api.asciidoc [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+[[clear-sql-cursor-api-request]]
+==== {api-request-title}
+
+`POST _sql/close`
+
+[[clear-sql-cursor-api-limitations]]
+===== Limitations
+
+See <<sql-limitations>>.
```
[the remark this answers is not in the document]
Comment (jrodewig) [reply docs/reference/sql/apis/clear-sql-cursor-api.asciidoc:41]: Thanks for pointing this out. This is largely here for future-proofing. I'll leave it in for now as it's consistent with our EQL docs.

**Comment by jrodewig** [comment]
Thanks, @bpintea!

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Review by bpintea — APPROVED** [review]
LGTM, thanks!

**Pull request merged**